src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -253,6 +253,86 @@ module.exports = class LevelPlane {
   }
 
   /**
+   * Determine whether or not the blocks at the given positions are powered
+   * rails that are connected to each other
+   *
+   * @param {Posititon} left
+   * @param {Posititon} right
+   * @return {boolean}
+   */
+  getPoweredRailsConnected(left, right) {
+    // return early if they're not even adjacent
+    if (!Position.isAdjacent(left, right)) {
+      return false;
+    }
+
+    const leftBlock = this.getBlockAt(left);
+    const rightBlock = this.getBlockAt(right);
+
+    // to be connected, both blocks must be powerable rails
+    if (!(leftBlock.getIsPowerableRail() && rightBlock.getIsPowerableRail())) {
+      return false;
+    }
+
+    // to be connected, both blocks must be oriented either North/South or
+    // East/West
+    const bothEastWest =
+      leftBlock.blockType.match('East|West') &&
+      rightBlock.blockType.match('East|West');
```

Comment:
Worth adding a `rightBlock.isHorizontal()` helper method?  A bit gross now when all block types live in the same class, but might make sense in the future when the different block types are separate classes.